gen/build_deploy/bash.py

```diff
@@ -361,6 +361,44 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
     fi
 }
 
+function d_type_enabled_if_xfs()
+{
+    # Return 1 if $1 is a directory on XFS volume with ftype ! = 1
+    # otherwise return 0
+    DIRNAME="$1"
+
+    RC=0
+    while [[ ! -d "$DIRNAME" ]]; do
+        DIRNAME="$(dirname "$DIRNAME")"
+    done
```

Comment:
This `while` loop could use a comment explaining what it's doing (removing any nonexistent portion of the path) and why (`df` will fail if the path doesn't exist).